bundle/config/mutator/resourcemutator/fix_permissions.go

```diff
@@ -121,6 +144,100 @@ func processPermissions(permissions dyn.Value, currentUser, resourceType string)
 	return dyn.V(permissionArray), nil
 }
 
+func useMaximumLevel(permissions dyn.Value, resourceType string) (dyn.Value, error) {
+	permissionArray, ok := permissions.AsSequence()
+	if !ok {
+		return permissions, nil
+	}
+
+	levelPerPrincipal := make(map[string]string)
+	principalIndex := make(map[string]int)
+	var principals []string
+
+	for _, permission := range permissionArray {
+		level, _ := dyn.GetValue(permission, "level").AsString()
+		if level == "" {
+			continue
+		}
+
+		principal := readPrincipal(permission)
+		if principal == "" {
+			continue
+		}
+		_, ok = principalIndex[principal]
+		if !ok {
+			ind := len(principalIndex)
+			principalIndex[principal] = ind
+			principals = append(principals, principal)
+		}
+		levelPerPrincipal[principal] = getMaxLevel(levelPerPrincipal[principal], level)
+	}
+
+	var newPermissions []dyn.Value
+
+	for _, principal := range principals {
+		newPermissions = append(newPermissions, createPermissionFromPrincipal(principal, levelPerPrincipal[principal]))
+	}
+
+	return dyn.V(newPermissions), nil
+}
+
+// Unified permission order map
+// Based on https://docs.databricks.com/aws/en/security/auth/access-control
+var PermissionOrder = map[string]int{
```

Comment:
What if a new permission level is introduced that is not in the list? What level score will it get? I assume `0`, right?